olive/passes/onnx/conversion.py

```diff
@@ -195,22 +219,95 @@ def _convert_model_on_device(
         # Reset to CPU so the resource consumed on GPU could be free.
         if device != "cpu":
             pytorch_model.to("cpu")
+
         if torch.cuda.is_available():
             torch.cuda.empty_cache()
-        # save the model to the output path and return the model
-        return model_proto_to_olive_model(onnx_model, output_model_path, config)
 
-
-class DeviceSpecificOnnxConversion(OnnxConversion):
-    """Convert a PyTorch model to ONNX model using torch.onnx.export by using specific hardware device."""
+        return onnx_model
 
     @staticmethod
-    def is_accelerator_agnostic(accelerator_spec: AcceleratorSpec) -> bool:
-        return False
+    def _export_ranked_model(params):
+        pass_config, data_root, model_config, world_size, device, local_rank, output_dirpath = params
+
+        import os
+
+        os.environ["OMPI_COMM_WORLD_RANK"] = str(local_rank)
+        os.environ["OMPI_COMM_WORLD_SIZE"] = str(world_size)
+        os.environ["OMPI_COMM_WORLD_LOCAL_RANK"] = str(local_rank)
+        os.environ["MIOPEN_FIND_MODE"] = "1"
+        os.environ["OMPI_MCA_btl"] = "^openib"  # noqa: SIM112
+        os.environ["OMPI_MCA_btl_openib_warn_no_device_params_found"] = "0"  # noqa: SIM112
+        os.environ["OMPI_MCA_pml"] = "ob1"  # noqa: SIM112
+        os.environ["OMPI_MCA_btl_tcp_if_include"] = "eth0"  # noqa: SIM112
+        os.environ["OMPI_MCA_hwloc_base_binding_policy"] = "numa"  # noqa: SIM112
+        os.environ["OMPI_MCA_ess"] = "^singleton"  # noqa: SIM112
+        os.environ["OMPI_MCA_ess_base_vpid"] = "0"  # noqa: SIM112
+        os.environ["OMPI_MCA_orte_tag_output"] = "1"  # noqa: SIM112
+        os.environ["OMPI_MCA_pmix"] = "^s1,s2,cray,isolated"  # noqa: SIM112
+        os.environ["OMPI_MCA_rmaps_ppr_n_pernode"] = "1"  # noqa: SIM112
+        os.environ["NCCL_DEBUG"] = "WARN"
+
+        from mpi4py.MPI import COMM_WORLD
+
+        world_size = COMM_WORLD.Get_size()
+        local_rank = COMM_WORLD.Get_rank()
+
+        torch.distributed.init_process_group(
+            "nccl", init_method="tcp://127.0.0.1:9876", world_size=world_size, rank=local_rank
+        )
+
+        output_filename = DistributedOnnxModel.DEFAULT_RANKED_MODEL_NAME_FORMAT.format(local_rank)
+        output_filepath = str(Path(output_dirpath) / output_filename)
+
+        input_model = DistributedPyTorchModel(**model_config)
+        pytorch_model = input_model.load_model(local_rank)
```

Comment:
maybe this can be called something like `olive_pytorch_model` to avoid confusion with `pytorch_model` which is of type `torch.nn.Module)

Reply:
Done!